### ISSUE TITLE
BUG: Frequent DOM Tree Updates / Re-rendering  or change events..

### ISSUE THREAD

**Issue by WeslleyDeSouza** [oblique-bit/oblique#141]
### Oblique version

11

### Browser & version

All

### Node version

-

### Device and operating system

-

### Is this a regression?

Yes

### Screenshots

![image](https://github.com/user-attachments/assets/cdf30004-c58e-4e5a-b03c-fd2998140565)

### Description

```true
The code is triggering DomTree Changes

 isInHighContrastMode() {
        const currentHighContrastMode = this.highContrastModeDetector.getHighContrastMode();
        return currentHighContrastMode === HighContrastMode.WHITE_ON_BLACK;
    }
```

When accessing the Oblique 11 Application and observing through DevTools, the body tag exhibits flickering behavior, which suggests continuous DOM manipulations. Additionally, upon manually deleting the body element from the DOM, the application generates an infinite loop of error messages. This behavior indicates potential issues with DOM stability and error handling mechanisms.
```


### Reproduction

```true
1. Open the Oblique 11 Application.
2. Launch DevTools. Click somewhere inside the body element.
3. Observe the body tag flickering, indicating ongoing DOM manipulations.
4. Remove the body element to observe an infinite stream of error messages.
```

### Expected behaviour

```true
No Dom Events, if nothing has been changed.
```

### Proposed solution

Best solution would be to prevent call the function again and again.. 
but for a workaround maybe you could store the result of the firstCall ?

**Comment by gillerr** [comment]
Hi @WeslleyDeSouza 

Thanks for the report, I was unaware of this problem.

We actually monitor High Contrast Mode changes with `ngAfterContentChecked` so that Oblique immediately reacts to any change to this setting. The way Angular detects that is by injecting a `div` in the `body` and then removing it. This is why the `body` is flickering and there is an endless loop of errors if you remove the `body`.

Now, this behavior trigger way too many change detection cycles, which is bad for the performances. The only solution that I can think of is to only check the High Contrast Mode on startup, but that will have an impact on accessibility. I will discuss this in my next meeting with Access For All on the 15. October.

**Comment by nina-egger** [comment]
fixed with 12.0.2
